### PR TITLE
Opets/us128971 use ou filter in ed

### PR DESCRIPTION
**Pull request by opets**
* Functional Testing
  * [x] Browsers (Chrome, FF, Edge)
  * [x] Accessibility
  * [x] RTL
  * [x] Responsive
  * [x] Test cases
* Security, devops, perf: N/A

### CODE REVIEW

**Comment by NicholasHallman** [comment]
Context? I thought @Vitalii-Misechko  already made these changes

File: model/data.js [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+		super();
+		this._data = data;
+	}
+
+	async fetchRelevantChildren(id, bookmark) {
+		return await fetchRelevantChildren(id, bookmark);
+	}
+
+	async orgUnitSearch(searchString, bookmark) {
+		return await orgUnitSearch(searchString, bookmark);
```
Comment (Vitalii-Misechko) [inline model/data.js:22]: you also need to pass `selectedSemesterIds`

File: model/data.js [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+ */
+class EngagementOuFilterDataManager extends OuFilterDataManager {
+
+	constructor(data) {
+		super();
+		this._data = data;
+	}
+
+	async fetchRelevantChildren(id, bookmark) {
+		return await fetchRelevantChildren(id, bookmark);
```
Comment (Vitalii-Misechko) [inline model/data.js:18]: you also need to pass `selectedSemesterIds`

File: model/data.js [hunk 3]
```diff
@@ -78,10 +107,13 @@ export class Data {
 	// @action
 	onServerDataReload(newServerData) {
 		const lastSearchResults = fetchLastSearch(newServerData.selectedSemestersIds);
+		const nodes = lastSearchResults ? [...newServerData.orgUnits, ...lastSearchResults] : newServerData.orgUnits;
+		const mappedNodes = nodes.map(n => ({ Id: n[0], Name: n[1], Type: n[2], Parents: n[3], IsActive: n[4] }));
```
Comment (Vitalii-Misechko) [inline model/data.js:111]: Could you use constants for array indeces?

**Comment by Vitalii-Misechko** [comment]
> Context? I thought @Vitalii-Misechko already made these changes

@NicholasHallman, I handed off this part to @opets.

File: model/data.js (hunk 3 above)
Comment (opets) [reply model/data.js:111]: I haven't found how to do this(
image
anyway, this is a temp fix, in the next PRs I will get structured data here and remove this code

Comment (Vitalii-Misechko) [reply model/data.js:111]: I meant to make something like this
```JS
const mappedNodes = nodes.map(n => ({ Id: n[ORG_UNIT.ID], Name: n[ORG_UNIT.NAME]...
```

**Review by johngwilkinson — APPROVED** [review]
LGTM, assuming those visual diff failures are just flaky tests.

**Comment by NicholasHallman** [comment]
> LGTM, assuming those visual diff failures are just flaky tests.

Yee flakes are due to vis-diff updates https://d2l.slack.com/archives/C0PHG3QB0/p1623867426284300

**Review by Vitalii-Misechko — APPROVED** [review]
LGTM!

**Comment by opets** [comment]
Hmm, I'm a bit confused😞 
looking to [skeleton diff](https://visual-diff.d2l.dev/screenshots/Brightspace/insights-engagement-dashboard/engagement-dashboard-skeleton/2021-06-19-13.27.12.963-f6e3d78cd8995e6562b0f758d9fbbcd5f8d88732-report.html), it seems the email button is gone(
image

but it exists on [the normal diff](https://visual-diff.d2l.dev/screenshots/Brightspace/insights-engagement-dashboard/engagement-dashboard/2021-06-19-13.27.18.533-f6e3d78cd8995e6562b0f758d9fbbcd5f8d88732-report.html) and in the mobile version also, 
on the local machine, I was also unable to reproduce this issue, my email button was drawn at start without any delay

**Comment by Vitalii-Misechko** [comment]
> Hmm, I'm a bit confused😞
> looking to [skeleton diff](https://visual-diff.d2l.dev/screenshots/Brightspace/insights-engagement-dashboard/engagement-dashboard-skeleton/2021-06-19-13.27.12.963-f6e3d78cd8995e6562b0f758d9fbbcd5f8d88732-report.html), it seems the email button is gone(
> image
> 
> but it exists in [the normal diff](https://visual-diff.d2l.dev/screenshots/Brightspace/insights-engagement-dashboard/engagement-dashboard/2021-06-19-13.27.18.533-f6e3d78cd8995e6562b0f758d9fbbcd5f8d88732-report.html) and in the mobile version also,
> on the local machine, I was also unable to reproduce this issue, my email button was drawn at start without any delay

It doesn't look good. You need to fix it. I guess adding delay might help. JFYI. We do not merge broken visual diffs
> await new Promise(resolve => setTimeout(resolve, 500));

**Comment by opets** [comment]
> It doesn't look good. You need to fix it. I guess adding delay might help. JFYI. We do not merge broken visual diffs

seems, I found the issue: [min-to-show="1"](https://github.com/Brightspace/insights-engagement-dashboard/pull/584/files#diff-da16422f38b47f7e8f1ecb041f786890b12adf07da4b62f2ab5a7d539b44a9e3R518)
there is no sense in specifying this value to 0, 'email' button will always be displayed.

**Pull request merged**